src/components/donation-flow/DonationFlowForm.tsx

```diff
@@ -89,82 +90,102 @@ export const validationSchema: yup.SchemaOf<DonationFormDataV2> = yup
-        authentication: session?.user ? DonationFormDataAuthState.AUTHENTICATED : null,
+        authentication: session?.user ? DonationFormAuthState.AUTHENTICATED : null,
       }}
       validationSchema={validationSchema}
       onSubmit={async (values) => {
+        if (values.payment === DonationFormPaymentMethod.BANK) {
+          router.push(routes.campaigns.donationStauts(campaign.slug), {
+            query: {
+              bank_payment: true,
+            },
+          })
+        }
+        setSubmitPaymentLoading(true)
         if (!stripe || !elements) {
           // Stripe.js has not yet loaded.
-          // Make sure to disable form submission until Stripe.js has loaded.
+          // Form should be disabled but TS doesn't know that.
+          setSubmitPaymentLoading(false)
           throw new Error('Stripe.js has not loaded when trying to submit the form')
         }
 
         if (!stripePaymentIntent) {
+          setSubmitPaymentLoading(false)
           throw new Error('Stripe payment intent does not exist when trying to submit the form')
         }
 
+        // Update the payment intent with the latest calculated amount
         try {
           await updatePaymentIntentMutation.mutateAsync({
             id: stripePaymentIntent.id,
             payload: {
               amount: Math.round(Number(values.finalAmount)),
               currency: campaign.currency,
+              metadata: {
+                campaignId: campaign.id,
+              },
             },
           })
         } catch (error) {
+          setSubmitPaymentLoading(false)
           setPaymentError({
             type: 'invalid_request_error',
             message: "We couldn't update the payment intent. Please try again later.",
           })
+          return
         }
 
+        // Create the payment entity
         try {
           await createStripePaymentMutation.mutateAsync({
             isAnonymous: values.isAnonymous,
             personEmail: session?.user?.email || values.email,
-            paymentIntentId: stripePaymentIntent?.id,
+            paymentIntentId: stripePaymentIntent.id,
             firstName: session?.user?.given_name || null,
             lastName: session?.user?.family_name || null,
             phone: null,
           })
         } catch (error) {
+          setSubmitPaymentLoading(false)
           setPaymentError({
             type: 'invalid_request_error',
             message: "We couldn't create the payment. Please try again later.",
           })
+          return
         }
 
+        // Confirm the payment
         const { error } = await stripe.confirmPayment({
           //`Elements` instance that was used to create the Payment Element
           elements,
           confirmParams: {
-            return_url: `${window.location.origin}/campaigns/donation-v2/${campaign.slug}`,
+            return_url: `${window.location.origin}/campaigns/donation-v2/${campaign.slug}/status`,
```

Comment:
Maybe we can use `routes` here instead of hardcoded links. It will ease us when we will update the routes and remove the `v2` suffix